realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -316,6 +316,11 @@ private static void initializeRealm(Realm realm) {
                 columnInfoMap.put(modelClass, mediator.validateTable(modelClass, realm.sharedGroupManager.getTransaction()));
             }
             realm.schema.columnIndices = new ColumnIndices(columnInfoMap);
+
+            final Transaction transaction = realm.getConfiguration().getInitialDataTransaction();
```

Comment:
@cmelchior Yes, you are right. It will be executed second time but without `realm.commitTransaction`. I think simple `version` check will do the job.